File: valid_perfect_square/best.js
// Runtime 72 ms, Memory Usage 33.8 mb

function validSquare(num) {
    // Built in JS function. First, Math.sqrt(num) will find the square root of the number.
    // Then, Number.isInteger() will determine if the number is whole, and therefore an integer.
    // Number.isInteger returns a boolean.
    return Number.isInteger(Math.sqrt(num))
};

File: valid_perfect_square/best.rb
# Runtime 44ms, Memory Usage 9.4 mb

def is_perfect_square(num)
    # Checks to see if the square root rounded down to the nearest whole number is the same.
    # If the number is a whole number, it should be equal.
    Math.sqrt(num) == Math.sqrt(num).floor
end